src/controllers/capp.go

```diff
@@ -178,6 +189,71 @@ func (c *cappController) GetCappState(namespace, name string) (types.GetCappStat
 	return cappState, nil
 }
 
+func (c *cappController) GetCappDNS(namespace, name string) (types.GetDnsResponse, error) {
+	c.logger.Debug(fmt.Sprintf("Trying to fetch dns related to capp %q in namespace %q", name, namespace))
+
+	recordsK8S := &dnsrecordv1alpha1.CNAMERecordList{}
+
+	listOptions := prepareDNSListOptions(namespace, name)
+
+	err := c.client.List(c.ctx, recordsK8S, listOptions)
+	if err != nil {
+		c.logger.Error(fmt.Sprintf("Could not fetch dns related to capp %q in namespace %q with error: %v", name, namespace, err.Error()))
+		return types.GetDnsResponse{}, err
+	}
+
+	if len(recordsK8S.Items) == 0 {
+		_, err := c.GetCapp(namespace, name)
+		if err != nil {
+			return types.GetDnsResponse{}, err
+		}
+		return types.GetDnsResponse{}, nil
+	}
+
+	return types.GetDnsResponse{Records: parseDNSList(recordsK8S)}, nil
+}
+
+// parseDnsList converts dns from k8s object to the response dns type.
+func parseDNSList(recordsK8S *dnsrecordv1alpha1.CNAMERecordList) []types.DNS {
```

Comment:
I believe we have different naming conventions to this type of functions, which I don't necessarily like better but wonder if you considered? 

https://github.com/dana-team/platform-backend/blob/d74366b2206529887b8c92a225306254a1ba0471/src/controllers/capp.go#L284

In any case maybe something like:

```go
// parseDNSList converts DNSRecords from their Kubernetes representation to a custom type representation.
```